src/execution/vm/bytecode_generator.cpp

```diff
@@ -1711,15 +1733,36 @@ void BytecodeGenerator::VisitBuiltinTrigCall(ast::CallExpr *call, ast::Builtin b
 }
 
 void BytecodeGenerator::VisitBuiltinArithmeticCall(ast::CallExpr *call, ast::Builtin builtin) {
-  LocalVar dest, src;
-  auto dest_type = call->GetType();
-  dest = GetExecutionResult()->GetOrCreateDestination(dest_type);
-  src = VisitExpressionForRValue(call->Arguments()[0]);
-  Bytecode abs_bytecode = call->Arguments()[0]->GetType()->IsIntegerType() ||
-                                  call->Arguments()[0]->GetType()->IsSpecificBuiltin(ast::BuiltinType::Integer)
-                              ? Bytecode::AbsInteger
-                              : Bytecode::AbsReal;
-  GetEmitter()->Emit(abs_bytecode, dest, src);
+  LocalVar dest = GetExecutionResult()->GetOrCreateDestination(call->GetType());
+
+  switch (builtin) {
+    case ast::Builtin::Abs: {
+      LocalVar src = VisitExpressionForRValue(call->Arguments()[0]);
+      Bytecode abs_bytecode = call->Arguments()[0]->GetType()->IsIntegerType() ||
+                                      call->Arguments()[0]->GetType()->IsSpecificBuiltin(ast::BuiltinType::Integer)
+                                  ? Bytecode::AbsInteger
+                                  : Bytecode::AbsReal;
+      GetEmitter()->Emit(abs_bytecode, dest, src);
+      break;
+    }
+    case ast::Builtin::Mod: {
+      LocalVar first_input = VisitExpressionForRValue(call->Arguments()[0]);
+      LocalVar second_input = VisitExpressionForRValue(call->Arguments()[1]);
+
+      bool first_input_is_int = call->Arguments()[0]->GetType()->IsIntegerType() ||
```

Comment:
Same here: remove checks for `IsIntegerType()` and just focus on SQL values.

Reply:
Can you copy the logic in `VisitSqlArithmeticExpr()` since it's effectively doing the same thing?

Reply:
Done